tests/package.json

```diff
@@ -0,0 +1,42 @@
+  "devDependencies": {
+    "cypress": "^12.17.2",
+    "cypress-plugin-snapshots": "^1.4.4",
+    "start-server-and-test": "^2.0.0"
+  },
+  "scripts": {
+    "installDependencies": "Rscript cypress/r_scripts/install_dependencies.R",
+    "circularLinearWithBrush": "Rscript cypress/r_scripts/run_circular_linear_with_brush.R",
+    "barChart": "Rscript cypress/r_scripts/run_barchart.R",
+    "areaChart": "Rscript cypress/r_scripts/run_area_chart.R",
+    "circularVisualEncoding": "Rscript cypress/r_scripts/run_circularVisualEncoding.R",
+    "ideogram": "Rscript cypress/r_scripts/run_ideogram.R",
+    "e2e-circularLinearWithBrush": "start-server-and-test circularLinearWithBrush http://localhost:8888 'cypress run --spec \"cypress/e2e/circularLinearWithBrushSpec.cy.js\" --record false'",
+    "e2e-barChart": "start-server-and-test barChart http://localhost:8888 'cypress run --spec \"cypress/e2e/barChart.cy.js\" --record false'",
+    "e2e-areaChart": "start-server-and-test areaChart http://localhost:8888 'cypress run --spec \"cypress/e2e/areaChart.cy.js\" --record false'",
+    "e2e-ideogram": "start-server-and-test ideogram http://localhost:8888 'cypress run --spec \"cypress/e2e/ideogram.cy.js\" --record false'",
+    "e2e-circularVisualEncoding": "start-server-and-test circularVisualEncoding http://localhost:8888 'cypress run --spec \"cypress/e2e/circularVisualEncoding.cy.js\" --record false'",
+    "multiTrackApp": "Rscript cypress/r_scripts/run_multiTrackApp.R",
+    "e2e-multiTrackApp": "start-server-and-test multiTrackApp http://localhost:8888 'cypress run --spec \"cypress/e2e/multiTrackApp.cy.js\" --record false'",
+    "pointPlot": "Rscript cypress/r_scripts/run_pointPlot.R",
+    "e2e-pointPlot": "start-server-and-test pointPlot http://localhost:8888 'cypress run --spec \"cypress/e2e/pointPlot.cy.js\" --record false'",
+    "ruleMarks": "Rscript cypress/r_scripts/run_ruleMarks.R",
+    "e2e-ruleMarks": "start-server-and-test ruleMarks http://localhost:8888 'cypress run --spec \"cypress/e2e/ruleMarks.cy.js\" --record false'",
+    "sarsCov2": "Rscript cypress/r_scripts/run_sarsCov2.R",
+    "e2e-sarsCov2": "start-server-and-test sarsCov2 http://localhost:8888 'cypress run --spec \"cypress/e2e/sarsCov2.cy.js\" --record false'",
+    "lineChart": "Rscript cypress/r_scripts/run_lineChart.R",
+    "e2e-lineChart": "start-server-and-test lineChart http://localhost:8888 'cypress run --spec \"cypress/e2e/lineChart.cy.js\" --record false'",
+    "staticCircularBar": "Rscript cypress/r_scripts/run_staticCircularBar.R",
+    "e2e-staticCircularBar": "start-server-and-test staticCircularBar http://localhost:8888 'cypress run --spec \"cypress/e2e/staticCircularBar.cy.js\" --record false'",
+    "structuralVariant": "Rscript cypress/r_scripts/run_structuralVariant.R",
+    "e2e-structuralVariant": "start-server-and-test structuralVariant http://localhost:8888 'cypress run --spec \"cypress/e2e/structuralVariant.cy.js\" --record false'",
```

Comment:
Can we have a minimal package json scripts?
Having a separate script for every app makes us add a new script over here everytimewe add a new app to example. It would be better to have fewer scripts such that it would test all the files. That script could trigger a script that runs a test for a given file so that people can still choose to run test for a single file by using that smaller script.

Reply:
Can you provide a minimal example? I'm not sure that I get the idea you're proposing
Isn't `run_cypress_test.sh` cover this case?

> That script could trigger a script that runs a test for a given file so that people can still choose to run test for a single file by using that smaller script.

This is already happening if you run `yarn e2e-structuralVariant` Or maybe I'm not fully understanding your comment here :slightly_smiling_face:

Reply:
The main suggestion is that whenever a new example is added nothing needs to be changed in these test suit files.

Reply:
Not sure how the code will look for this. But a similar logic was used for `teal.gallery` before where the [deploy script](https://github.com/insightsengineering/teal.gallery/blob/cadbe690c23fe8fd0aad9b8ef739e42e15f1493b/R/deploy.R) had a feature to deploy a specific app or all apps.
We could have similar two scripts in the `package.json` which would test one particular example by providing example name or run all examples. If that makes any sense. If this is still not clear I will provide you a better working example next week :)

Reply:
Oh well, we already have something like this but not a yarn script to run all of them. Will definitely add it.

We also have a small script to add simple snapshot tests automatically and yarn scripts automatically when we add an example app. [See here](https://github.com/Appsilon/shiny.gosling/blob/e58804a63193f2ef7f520722a4e05029f8dbfd88/tests/cypress/create_tests_templates.sh).

And also another script to update the script running all the tests. [See here](https://github.com/Appsilon/shiny.gosling/blob/e58804a63193f2ef7f520722a4e05029f8dbfd88/tests/cypress/update_run_all_tests.sh)

For example if you added the example to `inst/examples/myNewExample`
Then you need to run `create_tests_templates myNewExample` And this will create a new cypress spec for this app and also add the yarn script to the `package.json` file.
After that if you run `update_run_all_tests` this [file](https://github.com/Appsilon/shiny.gosling/blob/e58804a63193f2ef7f520722a4e05029f8dbfd88/tests/run_cypress_tests.sh) that is running all of the tests is updated with the latest yarn scripts added to that file.


On the other hand if you want to run the snapshot tests for only one of the apps then you just need to pick up one of the `e2e-appName` scripts in `package.json` and run `yarn e2e-appName` and this will run the e2e tests for only that app.

Reply:
Also I think that I understand another aspect of you comment as well which is (correct me if I'm wrong) to reduce the amount of entries in the `package.json` right? 
I guess that we can do it. I like the idea but the solution will definitely be tightly coupled with the implementation of e2e tests and writing of specs. So definitely I should write a small `README` guide on how to add new tests. I'll draft something

Reply:
@vedhav Please check #120